### PR TITLE
feat: implement INVALID opcode

### LINKED ISSUE #83: Opcode: INVALID

### PR DESCRIPTION
**Pull request by ricomateo**
Closes #83, #279
This PR implements code generation for the [INVALID](https://www.evm.codes/#fe?fork=cancun) opcode.

### CODE REVIEW

**Review by azteca1998 — APPROVED** [review]
You still have conflicts.

File: src/program.rs [hunk 1]
```diff
@@ -455,25 +436,18 @@ pub struct Program {
 }
 
 impl Program {
-    pub fn from_bytecode(bytecode: &[u8]) -> Result<Self, ParseError> {
+    pub fn from_bytecode(bytecode: &[u8]) -> Self {
```
Comment (MegaRedHand) [inline src/program.rs:439]: Let's keep the old version as `from_bytecode_checked`

File: src/program.rs [hunk 2]
```diff
@@ -808,6 +513,348 @@ impl Program {
...
+            .collect::<Vec<u8>>()
+    }
+
+    fn parse_operation(
+        bytecode: &[u8],
+        mut pc: usize,
+    ) -> Result<Option<(Operation, usize)>, OpcodeParseError> {
+        let Some(opcode) = bytecode.get(pc).copied() else {
+            return Ok(None);
+        };
```
Comment (MegaRedHand) [inline src/program.rs:555]: No need to return an option, since we already know this fetch cannot fail

**Review by MegaRedHand — APPROVED** [review]
LGTM. Left a comment, but it's minor.

**Pull request merged**